modules/globebrowsing/tile/rawtiledatareader/gdalrawtiledatareader.h

```diff
@@ -0,0 +1,121 @@
+#include <modules/globebrowsing/tile/tile.h>
+#include <modules/globebrowsing/tile/tiledepthtransform.h>
+#include <modules/globebrowsing/tile/tiledatalayout.h>
+#include <modules/globebrowsing/tile/pixelregion.h>
+#include <modules/globebrowsing/tile/rawtile.h>
+
+#include <modules/globebrowsing/tile/rawtiledatareader/rawtiledatareader.h>
+
+#include <ghoul/glm.h>
+#include <ghoul/opengl/ghoul_gl.h>
+#include <ghoul/opengl/texture.h>
+
+#include <gdal.h>
+#include <string>
+
+class GDALDataset;
+class GDALRasterBand;
+
+namespace openspace {
+namespace globebrowsing {
+
+class GeodeticPatch;
+
+class GdalRawTileDataReader : public RawTileDataReader {
+public:
+
+    /**
+    * Opens a GDALDataset in readonly mode and calculates meta data required for 
+    * reading tile using a TileIndex.
+    *
+    * \param filePath, a path to a specific file GDAL can read
+    * \param config, Configuration used for initialization
+    */
+    GdalRawTileDataReader(const std::string& filePath, const Configuration& config);
+
+    virtual ~GdalRawTileDataReader() override;
+
+    // Public virtual function overloading
+    virtual void reset() override;
+    virtual int maxChunkLevel() override;
+    virtual float noDataValueAsFloat() const override;
+    virtual int rasterXSize() const override;
+    virtual int rasterYSize() const override;
+    virtual float depthOffset() const override;
+    virtual float depthScale() const override;
+
+protected:
+
+    /**
+     * Returns the geo transform from raster space to projection coordinates as defined
+     * by GDAL.
+     * If the transform is not available, the function returns a transform to map
+     * the pixel coordinates to cover the whole geodetic lat long space.
+    */
+    virtual std::array<double, 6> getGeoTransform() const override;
+    virtual IODescription getIODescription(const TileIndex& tileIndex) const override;
+
+private:
+    // Private virtual function overloading
+    virtual void initialize() override;
+    virtual char* readImageData(
+        IODescription& io, RawTile::ReadError& worstError) const override;
+    virtual RawTile::ReadError rasterRead(
+        int rasterBand, const IODescription& io, char* dst) const override;
+
+    // GDAL Helper methods
+    GDALDataset* openGdalDataset(const std::string& filePath);
+    int calculateTileLevelDifference(int minimumPixelSize);
+    bool gdalHasOverviews() const;
+    int gdalOverview(const PixelRegion::PixelRange& baseRegionSize) const;
+    int gdalOverview(const TileIndex& tileIndex) const;
+    int gdalVirtualOverview(const TileIndex& tileIndex) const;
+    PixelRegion gdalPixelRegion(GDALRasterBand* rasterBand) const;
+    TileDataLayout getTileDataLayout(GLuint prefferedGLType);
+
+    // Member variables
+    struct InitData {
+        std::string initDirectory;
+        std::string datasetFilePath;
+        int tilePixelSize;
+        GLuint dataType;
+    } _initData;
+
+    GDALDataset* _dataset;
```

Comment:
GDAL does not like that. The object is created in GDALOpen and destroyed in GDALClose. That does not play well with unique_ptr which tries to delete the object after GDALClose has been called.